app/components/primer/dropdown.rb

```diff
@@ -0,0 +1,134 @@
+    # @param direction [Symbol] <%= one_of(Primer::Dropdown::Menu::DIRECTION_OPTIONS) %>
+    # @param scheme [Symbol] Pass `:dark` for dark mode theming
+    # @param header [String] Optional string to display as the header
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :menu, "Primer::Dropdown::Menu"
+
+    # @example Default
+    #   <%= render(Primer::Dropdown.new) do |c| %>
+    #     <% c.button do %>
+    #       Dropdown
+    #     <% end %>
+    #
+    #     <%= c.menu(header: "Options") do |menu|
+    #       menu.item { "Item 1" }
+    #       menu.item { "Item 2" }
+    #       menu.item(divider: true)
+    #       menu.item { "Item 3" }
+    #       menu.item { "Item 4" }
+    #     end %>
+    #   <% end %>
+    #
+    # @example With direction
+    #   <%= render(Primer::Dropdown.new(display: :inline_block)) do |c| %>
+    #     <% c.button do %>
+    #       Dropdown
+    #     <% end %>
+    #
+    #     <%= c.menu(header: "Options", direction: :s) do |menu|
+    #       menu.item { "Item 1" }
+    #       menu.item { "Item 2" }
+    #       menu.item(divider: true)
+    #       menu.item { "Item 3" }
+    #       menu.item { "Item 4" }
+    #     end %>
+    #   <% end %>
+    #
+    # @example With caret
+    #   <%= render(Primer::Dropdown.new(with_caret: true)) do |c| %>
+    #     <% c.button do %>
+    #       Dropdown
+    #     <% end %>
+    #
+    #     <%= c.menu(header: "Options") do |menu|
+    #       menu.item { "Item 1" }
+    #       menu.item { "Item 2" }
+    #       menu.item(divider: true)
+    #       menu.item { "Item 3" }
+    #       menu.item { "Item 4" }
+    #     end %>
+    #   <% end %>
+    #
+    # @example Customizing the button
+    #   <%= render(Primer::Dropdown.new) do |c| %>
+    #     <% c.button(scheme: :primary, variant: :small) do %>
+    #       Dropdown
+    #     <% end %>
+    #
+    #     <%= c.menu(header: "Options") do |menu|
+    #       menu.item { "Item 1" }
+    #       menu.item { "Item 2" }
+    #       menu.item(divider: true)
+    #       menu.item { "Item 3" }
+    #       menu.item { "Item 4" }
+    #     end %>
+    #   <% end %>
+    #
+    # @example Menu as list
+    #   <%= render(Primer::Dropdown.new) do |c| %>
+    #     <% c.button do %>
+    #       Dropdown
+    #     <% end %>
+    #
+    #     <%= c.menu(as: :list, header: "Options") do |menu|
+    #       menu.item { "Item 1" }
+    #       menu.item { "Item 2" }
+    #       menu.item(divider: true)
+    #       menu.item { "Item 3" }
+    #       menu.item { "Item 4" }
+    #     end %>
+    #   <% end %>
+    #
+    # @example Customizing menu items
+    #   <%= render(Primer::Dropdown.new) do |c| %>
+    #     <% c.button do %>
+    #       Dropdown
+    #     <% end %>
+    #
+    #     <%= c.menu(header: "Options") do |menu|
+    #       menu.item(tag: :summary) { "Item 1" }
```

Comment:
I think we might want to leave `summary` out as a menu item since it should rarely/never be used.